// hello.proto
syntax = "proto3";

option java_multiple_files = true;
option java_package = "pb.java.microservices.grpc.profile.generatedProto";
package profile;

service Profile {
  rpc GetProfiles(Request) returns (Result);
}

message Request {
  repeated string hotelIds = 1;
}

message Result {
  repeated Hotel hotels = 1;
}

message Hotel {
  string id = 1;
  string name = 2;
  string phoneNumber = 3;
  string description = 4;
  Address address = 5;
  repeated Image images = 6;
}

message Address {
  string streetNumber = 1;
  string streetName = 2;
  string city = 3;
  string state = 4;
  string country = 5;
  string postalCode = 6;
  float lat = 7;
  float lon = 8;
}

message Image {
  string url = 1;
  bool default = 2;
}